From 4ac5726747a3f272403b674ab9d2a3f4f01be3fa Mon Sep 17 00:00:00 2001
From: Andy Grover <agrover@redhat.com>
Date: Sat, 10 Aug 2013 15:40:57 -0700
Subject: [PATCH 04/17] iser-Fix-bidirectional-chap

---
 usr/iscsi/iser_text.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/usr/iscsi/iser_text.c b/usr/iscsi/iser_text.c
index 5580420..9595d09 100644
--- a/usr/iscsi/iser_text.c
+++ b/usr/iscsi/iser_text.c
@@ -601,6 +601,8 @@ void iser_login_exec(struct iscsi_connection *iscsi_conn,
 	struct iscsi_login_rsp *rsp_bhs = (struct iscsi_login_rsp *)tx_pdu->bhs;
 	int stay = 0, nsg_disagree = 0;
 
+	tx_pdu->membuf.size = 0;
+
 	memset(rsp_bhs, 0, BHS_SIZE);
 	if ((req_bhs->opcode & ISCSI_OPCODE_MASK) != ISCSI_OP_LOGIN ||
 	    !(req_bhs->opcode & ISCSI_OP_IMMEDIATE)) {
-- 
1.7.1

